src/internal/got/client.ts

```diff
@@ -87,7 +89,7 @@ interface RawAppGrants {
 export class GotCFToolsClient implements CFToolsClient {
     private readonly auth?: AuthorizationProvider;
 
-    constructor(private client: HttpClient, private serverApiId?: ServerApiId, auth?: AuthorizationProvider) {
+    constructor(private client: HttpClient, private serverApiId?: ServerApiId, private enterpriseToken?: string, private hasAccountCreationAccess?: boolean, auth?: AuthorizationProvider) {
```

Comment:
The client should never ever need to have access to, nor even the knowledge of an enterprise token. The token is authentication information only, the client should not care about that :)